### DISCUSSION TITLE
Вопросы по Creating an AceBase server

### DISCUSSION THREAD

**Discussion by LebedevIV** [appy-one/acebase-server#75] [Q&A] [answered]
Good time of the year!
I'm confused, lack experience.
I'm trying to set up AceBase server. According to the instructions, you need to set the authentication settings, an example of the settings:
const settings = {
    host: 'mydb.example.com',
    port: 80,
    authentication: {
        enabled: true,
        allowUserSignup: false,
        defaultAccessRule: 'auth',
        defaultAdminPassword: '75sdDSFg37w5'
    }
}
The settings are then applied in a call to const server = new AceBaseServer(dbname, settings);
This call happens in ./node_modules/acebase-server/dist/esm/start.js
This happens along the chain written in ./node_modules/acebase-server/package.json :
"npm run test" runs "npm run start" which runs with parameters
./node_modules/acebase-server/dist/esm/start.js 
AUTH is passed as one of the parameters, which sets
if (+getVariable('AUTH', 1) === 0) {
    options.authentication = { enabled: false };
}
And then there is a connection const server = new AceBaseServer(dbname, options);
In this case, allowUserSignup, defaultAccessRule and defaultAdminPassword in ./node_modules/acebase-server/dist/esm/start.js are not passed, and their values are taken from the default values from /root/node_modules/acebase-server/dist/esm/settings/index.js
./node_modules/acebase-server/dist/esm/start.js does not provide at all for receiving the allowUserSignup, defaultAccessRule and defaultAdminPassword parameters that could be passed from ./node_modules/acebase-server/package.json
Question:
How can allowUserSignup, defaultAccessRule be set ? Do you really need to edit ./node_modules/acebase-server/dist/esm/start.js ?
And if the server is already created - can I run it with the new parameters and they will be applied and override the old ones ?

Thank you!

**Comment by appy-one** [comment, marked as answer]
You are looking at 2 different ways to start up an AceBase server.
1. Through code: `const server = new AceBaseServer(dbname, options)`. This is the documented and preferred way to start up a server.
2. From the command line: `npm run test` or `npm run start` or `node ./dist/esm/start.js .DBNAME=mydb ....`. This allows starting up a server using command line arguments and/or environment variables to configure it. Not all options are available this way.
